pkg/device/device-util.go

```diff
@@ -85,12 +85,33 @@ func CreateVolume(vol *apis.DeviceVolume) error {
 		klog.Errorf("findBestPart Failed")
 		return err
 	}
-	return createPart(disk, start, partitionName, capacityMiB)
+	return wipefsAndCreatePart(disk, start, partitionName, capacityMiB, diskMetaName)
 }
 
-func createPart(disk string, start uint64, partitionName string, size uint64) error {
+// DeletePart Todo
+func wipefsAndCreatePart(disk string, start uint64, partitionName string, size uint64, diskMetaName string) error {
+	klog.Infof("Creating Partition %s %s", partitionName, diskMetaName)
 	_, err := RunCommand(strings.Split(fmt.Sprintf(PartitionCreate, disk, partitionName, start, start+size), " "))
-	return err
+	if err != nil {
+		klog.Errorf("Create Partition failed %s", err)
+		return err
+	}
+
+	pList, err := getAllPartsUsed(diskMetaName, partitionName)
+	if err != nil {
+		klog.Errorf("GetAllPartsUsed failed %s", err)
+		return err
+	}
+
+	err = wipeFsPartition(pList[0].DiskName, pList[0].PartNum)
+	if err != nil {
+		klog.Infof("Deleting partition %d on disk %s because wipefs failed", pList[0].PartNum, pList[0].DiskName)
+		_ = deletePartition(pList[0].DiskName, pList[0].PartNum)
```

Comment:
can we add an error message here, so that we know that leaked partition is because of this

Reply:
or you can go ahead with this, I will add as part of my refactoring

Reply:
Done. @praveengt . Added the log message.